src/libs/actions/Report.js

```diff
@@ -869,7 +971,12 @@ function handleReportChanged(report) {
     // A report can be missing a name if a comment is received via pusher event
     // and the report does not yet exist in Onyx (eg. a new DM created with the logged in person)
     if (report.reportID && report.reportName === undefined) {
-        fetchChatReportsByIDs([report.reportID]);
+        updateIOUAndChatReportData(report.reportID);
```

Comment:
Yeah, we should solve the problem of IOU reports being seen as chat reports at the place it occurs, rather than filtering them out here.

Reply:
When a new IOU report is created we receive a push notification for CREATED action with IOU report ID. 
We call updateReportWithNewAction and set Onyx object for report.

When we get a push notification with CREATED action, we don't know if it's an IOU or Chat report until we call API to fetch reportStuff.


It would be great if we fix the push notifications in the first place, so we don't have to reuse chatReport related functions and put if conditions.

Reply:
Ah, thanks for pointing out the problem. I looked into the data to confirm and it seems that we are returning the wrong action data (we're sending the very first action, instead of the latest IOU action).

Sorry that we didn't notice this earlier. I have created an internal issue so that we can unblock you asap. Perhaps there is a workaround via looking up the report after the notification is received, but I would like to avoid this and focus on fixing the actual issue.

Reply:
There is another push notification issue btw. When an IOU report is created, we don't get a new action push for the chat report, so we cannot display "Requested £.... for ...." quoted message until the page is refreshed.

Reply:
I'll look into that also. Thanks.

Reply:
Hi @tugbadogan, we've fixed both of these issues but it will take a couple of days to deploy to production. I'll let you know as soon as this has happened.

Reply:
Hey again @tugbadogan. Ignore my last message, the fix should now be live on production 🎉 

Please let us know if you have any further issues here.

Reply:
It looks like we are getting push notifications for just chat reports 👍 I updated PR and separated IOU update functions from chat report functions. We are not creating `report_<>` for IOU reports and we don't have to modify `handleReportChanged` function anymore.